Assets/Scripts/AchievementsService.cs

```diff
@@ -558,14 +583,27 @@ private async void GetAndCacheData(string uri)
                 await System.Threading.Tasks.Task.Yield();
             }
```

Comment:
This while loop should be removed, as the handling of the `completed` event obviates the need for it.